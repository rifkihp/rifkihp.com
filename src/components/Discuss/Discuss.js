// @flow
import { graphql } from 'gatsby';
import React from 'react';

import styles from './Discuss.module.scss';

type Props = {|
  +twitter?: string,
  +hn?: string,
  +reddit?: string,
|};

function openLink(link) {
  window.open(link);
}

export default ({ twitter, hn, reddit }: Props) =>
  twitter || hn || reddit ? (
    <div className={styles['discuss']}>
      <h4 className={styles['discuss-title']}>DISCUSS ON</h4>
      <div className={styles['discuss-buttons']}>
        {twitter && (
          <button
            className={styles['discuss-button-twitter']}
            onClick={openLink.bind(null, twitter)}
          >
            Twitter
          </button>
        )}
        {hn && (
          <button className={styles['discuss-button-hn']} onClick={openLink.bind(null, hn)}>
            Hacker News
          </button>
        )}
        {reddit && (
          <button className={styles['discuss-button-reddit']} onClick={openLink.bind(null, reddit)}>
            Reddit
          </button>
        )}
      </div>
    </div>
  ) : null;

export const fragment = graphql`
  fragment DiscussFragment on MarkdownRemarkFrontmatter {
    discussLinkTwitter
    discussLinkHN
    discussLinkReddit
  }
`;
