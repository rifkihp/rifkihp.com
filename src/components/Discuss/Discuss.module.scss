@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.discuss {
  @include margin-top(0.75);

  &-title {
    margin: 0;
    font-size: $typographic-base-font-size * 1;
    font-weight: bold;
    text-align: center;
  }

  &-buttons {
    display: flex;
    flex-direction: row;
    justify-content: center;

    button {
      width: 100%;
      flex-basis: 25%;
      padding: 10px 20px;
      text-align: center;
      white-space: nowrap;
      color: white; // don't invert
    }

    .discuss-button {
      &-hn {
        background-color: $color-hn;

        &:hover {
          background-color: darken($color-hn, 10%);
        }
      }

      &-twitter {
        background-color: $color-twitter;

        &:hover {
          background-color: darken($color-twitter, 10%);
        }
      }

      &-reddit {
        background-color: $color-reddit;

        &:hover {
          background-color: darken($color-reddit, 10%);
        }
      }
    }
  }
}

@include breakpoint-sm {
  .discuss {
    @include margin-top(1);

    &-buttons button {
      padding: 15px 30px;
    }
  }
}
