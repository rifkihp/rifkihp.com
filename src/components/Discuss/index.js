// @flow
export { default } from './Discuss';
