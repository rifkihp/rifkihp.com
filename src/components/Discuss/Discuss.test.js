import React from 'react';
import renderer from 'react-test-renderer';

import Discuss from './Discuss';

describe('Discuss', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Discuss
          twitter="https://twitter.com"
          hn="https://news.ycombinator.com"
          reddit="https://reddit.com"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
