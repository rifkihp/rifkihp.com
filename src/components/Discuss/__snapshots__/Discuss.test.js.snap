// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Discuss renders correctly 1`] = `
<div
  className="discuss"
>
  <h4
    className="discuss-title"
  >
    DISCUSS ON
  </h4>
  <div
    className="discuss-buttons"
  >
    <button
      className="discuss-button-twitter"
      onClick={[Function]}
    >
      Twitter
    </button>
    <button
      className="discuss-button-hn"
      onClick={[Function]}
    >
      Hacker News
    </button>
    <button
      className="discuss-button-reddit"
      onClick={[Function]}
    >
      Reddit
    </button>
  </div>
</div>
`;
